core/src/main/java/org/elasticsearch/http/HttpServer.java

```diff
@@ -42,25 +43,18 @@
  * A component to serve http requests, backed by rest handlers.
  */
 public class HttpServer extends AbstractLifecycleComponent<HttpServer> implements HttpServerAdapter {
-
-    private final Environment environment;
-
     private final HttpServerTransport transport;
 
     private final RestController restController;
 
     private final NodeService nodeService;
 
     @Inject
-    public HttpServer(Settings settings, Environment environment, HttpServerTransport transport,
-                      RestController restController,
-                      NodeService nodeService) {
+    public HttpServer(Settings settings, HttpServerTransport transport, RestController restController, NodeService nodeService) {
         super(settings);
-        this.environment = environment;
         this.transport = transport;
         this.restController = restController;
         this.nodeService = nodeService;
-        nodeService.setHttpServer(this);
```

Comment:
I'll undo the change. As a consequence, the setup code in `HttpServerTest` will get more complex.